src/client/Microsoft.Identity.Client/Internal/Requests/ClientCredentialRequest.cs

```diff
@@ -75,7 +75,11 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
 
                         SilentRequestHelper.ProcessFetchInBackground(
                         cachedAccessTokenItem,
-                        () => GetAccessTokenAsync(cancellationToken, logger), logger);
+                        () =>
+                        {
+                            using var tokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
```

Comment:
Pls add some comments on why this is done, i.e. to avoid the app disposing of the ct. Or have a helper method that is named appropriately.